stop floor under shadow statue opening immediately

briaguya0 opened the pull request:
fixes https://github.com/HarbourMasters/Shipwright/issues/805

`sStatueRotY` wasn't being reset, leading to the angle check

https://github.com/HarbourMasters/Shipwright/blob/235cef6abffea3fff1ec28fa4a7856464ec886bd/soh/src/overlays/actors/ovl_Bg_Haka_Gate/z_bg_haka_gate.c#L240

 in `FloorClosed` immediately passing

https://github.com/HarbourMasters/Shipwright/blob/235cef6abffea3fff1ec28fa4a7856464ec886bd/soh/src/overlays/actors/ovl_Bg_Haka_Gate/z_bg_haka_gate.c#L227-L257

by resetting `sStatueRotY` on init, this PR makes the statue and floor behave as expected

Commit:
stop floor under shadow statue opening immediately

dcvz reviewed (approved):
You're on a roll!

Kenix3 reviewed (changes requested):
Sbouldnt this be on the reset function rather than init?

briaguya0 commented:
>  Sbouldnt this be on the reset function

should i add a reset function? `z_bg_haka_gate.c` doesn't have one

Kenix3 commented:
> > Sbouldnt this be on the reset function
> 
> should i add a reset function? `z_bg_haka_gate.c` doesn't have one

Yes, please add one.

Commit:
use ActorResetFunc to reset sStatueRotY

briaguya0 commented:
> > > Sbouldnt this be on the reset function
> > 
> > 
> > should i add a reset function? `z_bg_haka_gate.c` doesn't have one
> 
> Yes, please add one.

added, tested, updated PR
